src/components/organisms/ProfileModal/UserInformationContent/UserInformationContent.tsx

```diff
@@ -1,179 +1,214 @@
-      profile.kidsMode = !profile?.kidsMode;
-      await updateUserProfile(user.uid, { kidsMode: profile.kidsMode });
+    if (user) {
+      user.kidsMode = !user?.kidsMode;
+      await updateUserProfile(user.id, { kidsMode: user.kidsMode });
     }
-  }, [profile, user]);
+  }, [user]);
 
   const toggleAnonMode = useCallback(async () => {
-    if (user && profile) {
-      profile.anonMode = !profile?.anonMode;
-      await updateUserProfile(user.uid, { anonMode: profile.anonMode });
+    if (user) {
+      user.anonMode = !user?.anonMode;
+      await updateUserProfile(user.id, { anonMode: user.anonMode });
     }
-  }, [profile, user]);
+  }, [user]);
 
   const toggleMirrorVideo = useCallback(async () => {
-    if (user && profile) {
-      profile.mirrorVideo = !profile?.mirrorVideo;
-      await updateUserProfile(user.uid, { mirrorVideo: profile.mirrorVideo });
+    if (user) {
+      user.mirrorVideo = !user?.mirrorVideo;
+      await updateUserProfile(user.id, { mirrorVideo: user.mirrorVideo });
     }
-  }, [profile, user]);
-
-  const renderedProfileQuestionAnswers = useMemo(
-    () =>
-      profile
-        ? profileQuestions?.map((question) => {
-            // @ts-ignore User type doesn't accept string indexing. We need to rework the way we store answers to profile questions
-            const questionAnswer = profile[question.name];
-
-            if (!questionAnswer) return undefined;
-
-            return (
-              <div key={question.name} className="question-section">
-                <div className="question">{question.text}</div>
-                <div className="answer">{questionAnswer}</div>
-              </div>
-            );
-          })
-        : undefined,
-    [profile, profileQuestions]
+  }, [user]);
+
+  const editProfileLink = useCallback(
+    (profileLink) => () => {
+      setProfileLinkToEdit(profileLink);
+      setUserProfileMode(UserProfileMode.EDIT_PROFILE_LINK);
+    },
+    [setProfileLinkToEdit, setUserProfileMode]
   );
 
-  if (!user || !userWithId) return null;
+  if (!user) return null;
 
   return (
-    <>
-      <h1 className="title modal-title">My Profile</h1>
-
-      <div className="user-information">
-        <UserAvatar user={userWithId} showStatus large />
-        <div className="text-container">
-          <h3>{profile?.partyName || DEFAULT_PARTY_NAME}</h3>
-          <div className="ellipsis-text">{user.email}</div>
-          <UserStatusDropdown />
+    <div className="UserInformationContent">
+      <h1 className="UserInformationContent__title">My Profile</h1>
+      <div className="UserInformationContent__information">
+        <div>
+          <UserAvatar user={user} showStatus large />
         </div>
+        <div className="UserInformationContent__text-container">
+          <h3 className="UserInformationContent__user-name">
+            {user?.partyName ?? DEFAULT_PARTY_NAME}
+          </h3>
+          <div
+            title={email ?? ""}
```

Comment:
Any reason to have `?? ""`?

Reply:
No point, if `email` is `undefined` then the prop will not be set, which is more correct/valid than setting it to an empty string. This is a blocker, please change.

Reply:
agree, there is no point  for `string | undefined` type. But we have `null` allowed. Should I remove it and leave just `string`?
```ts
export interface UserInfo {
  displayName: string | null;
  email: string | null;
  phoneNumber: string | null;
  photoURL: string | null;
  providerId: string;
  uid: string;
}
```

> Type 'null' is not assignable to type 'string | undefined' .ts(2322)

Reply:
I guess it's legacy then. Just leave a comment

Reply:
added a `@debt` comment https://github.com/sparkletown/sparkle/pull/1499/commits/354d8df4363f9df93752628de4e030466872f219